ddtrace/contrib/botocore/patch.py

```diff
@@ -335,6 +348,9 @@ def patched_api_call(original_func, instance, args, kwargs):
 
     endpoint_name = deep_getattr(instance, "_endpoint._endpoint_prefix")
 
+    if _PATCHED_SUBMODULES and endpoint_name not in _PATCHED_SUBMODULES:
+        return original_func(*args, **kwargs)
+
```

Comment:
It seems that we skip patching only when `_PATCHED_SUBMODULES` has modules to patch and the endpoint_name is not in those modules.

What is the point of having `patch_submodules(True)`?

Reply:
Good point, it is redundant and can be removed because `patch_submodules(True)` will execute the exact logic as calling `patch()`.
I can remove this part if you prefere:
```
    if isinstance(submodules, bool) and submodules:
        _PATCHED_SUBMODULES.clear()
```